Adds uui-box examples

krebil opened the pull request:
I've tried my hand at implementing https://github.com/umbraco/UI-Examples/issues/32

As umb-box is not a thing anymore I've added examples for uui-box instead. I'm not sure if there is enough descriptive text or not. 

Let me know about any adjustments you would like me to make.

fixes https://github.com/umbraco/UI-Examples/issues/32

krebil committed:
Adds uui-box examples

Rockerby commented on src/Our.Umbraco.UiExamples.v14/src/scripts/box/box-dashboard.ts:
```diff
@@ -0,0 +1,263 @@
+        return html`
+            <uui-box class="red-outline" .headline="test">
+                <h3 slot="headline" class="blue-outline spacing headline">
+                    The uui-box element
```
Consider updating this with the same text as in the HTML example code ("this is a headline") to make the example directly tie into what the user sees

Rockerby commented on src/Our.Umbraco.UiExamples.v14/src/scripts/box/box-dashboard.ts:
```diff
@@ -0,0 +1,263 @@
+                    The uui-box element
+                </h3>
+                <div slot="header" class="header purple-outline spacing">
+                    Other header content
```
Consider updating this with the same text as in the HTML example code ("this is a header") to make the example directly tie into what the user sees

Rockerby commented on src/Our.Umbraco.UiExamples.v14/src/scripts/box/box-dashboard.ts:
```diff
@@ -0,0 +1,263 @@
+                        <umb-code-block language="HTML">${this.renderBoxCodeExample()}</umb-code-block>
+                        <div>
+                            <p>
+                                The uui-box has largely replaces the umb-box element form previous versions of Umbraco.
```
Typo; `form` -> `from`

Rockerby commented on src/Our.Umbraco.UiExamples.v14/src/scripts/box/box-dashboard.ts:
```diff
@@ -0,0 +1,263 @@
+                    <umb-code-block language="HTML">${this.renderHeaderSlotCodeExample()}</umb-code-block>
+                    <p>
+                        The header slot is optional and is used to display additional content in the header of the box. 
+                    </p>
```
The header slot can also be added via the `header` attribute (like `headline`). Can we add an example of it here too please?

krebil replied:
Actually the only properties on uui-box is headline and headline variant, at least according to the storybook https://uui.umbraco.com/?path=/docs/uui-box--docs

Rockerby replied:
Very correct! Apologies, I was convinced that was an attribute tag as well 😊

Rockerby commented on src/Our.Umbraco.UiExamples.v14/src/scripts/box/box-dashboard.ts:
```diff
@@ -0,0 +1,263 @@
+        return html`
+&lt;uui-box&gt;
+    &lt;h3 slot=&quot;headline&quot;&gt;
+        this is a title
```
Consider rewording to `this is a headline`

Rockerby reviewed (changes requested):
Hey there @krebil ! Thanks for your contribution to the package 🙌

Great work on this, I think the new page showcases the UUI Box component well. I've added a few comments for you to take a look over before we get this merged in.

krebil committed:
Update text based on comments

krebil commented:
@Rockerby I've updated the text and corrected typos based on your comments. I haven't been able to find any documentation on uui-box with a header property and I can't get it to work when i try myself either. So I haven't added an example for that just yet.

Rockerby commented:
Thanks @krebil . That looks spot on, changes checked and merged #h5yr!
